### DISCUSSION TITLE
How to count how many records you have in a table?

### DISCUSSION THREAD

**Discussion by ReinaldoSergio** [simolus3/drift#1034] [General]
I'm using moor_flutter and I don't know how to count how many records are in a table, can someone tell me how to do this?

**Comment by simolus3** [comment]
[Counting](https://moor.simonbinder.eu/docs/getting-started/expressions/#counting) is explained in the documentation. You can use this:

```dart
Future<int> rowsIn(TableInfo table) {
  final amount = countAll();
  final query = selectOnly(table)..addColumns([amount]);
  return query.map((row) => row.read(amount)).getSingle();
}
```

**Reply by ReinaldoSergio** [reply]
I want to take this value and put it in an integer variable, but it's giving the following error: Future <int> from Function 'rowsIn'

**Reply by simolus3** [reply]
Well, you'll have to `await` the future to get an `int`

**Reply by ReinaldoSergio** [reply]
it worked, thank you very much
